### PR TITLE
Add libxc

### PR DESCRIPTION
**Pull request by bjodah**
Originally from https://github.com/psi4/psi4meta/tree/master/conda-recipes/libxc/

Dependency for gpaw (and many other ab initio packages). (cf. #7834)

EDIT: there is also: https://github.com/abinit/abiconda/tree/master/recipes/libxc

### CODE REVIEW

**Comment by bjodah** [comment]
Looks like macOS failures might be due to tight tolerances (but I don't see any unsafe math optimization flags passed to the compiler). The windows failure looks like it might need a later version of vc.

**Comment by bjodah** [comment]
ready for review

File: recipes/libxc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  #url: http://www.tddft.org/programs/octopus/down.php?file=libxc/4.3.3/libxc-4.3.3.tar.gz
+  #sha256: 99ed7a31c7ae19e364d341e2d4a02b108a9b4f45c7f5c5926c6108612784aff0
+  url: https://gitlab.com/libxc/libxc/-/archive/{{ version }}/libxc-{{ version }}.tar.bz2
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [not linux]
+  run_exports:
+    - {{ pin_subpackage('libxc') }}
```
Comment (chrisburr) [inline recipes/libxc/meta.yaml]: Is the ABI of libxc stable for all major versions? It's more common to be stable for minor versions (if at all) which requires one of:
```yaml
  - {{ pin_subpackage('libxc', max_pin='x.x') }}
  - {{ pin_subpackage('libxc', max_pin='x.x.x') }}
```

File: recipes/libxc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{% set name = "libxc" %}
+{% set version = "4.3.3" %}
+{% set sha256 = "4dedcadbb5755b6da1836effabf0ad0d77a1b6fb6760d847a456d1ace9145faa" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  #url: http://www.tddft.org/programs/octopus/down.php?file=libxc/4.3.3/libxc-4.3.3.tar.gz
```
Comment (chrisburr) [inline recipes/libxc/meta.yaml]: These comments should be deleted I think?

Comment (bjodah) [reply recipes/libxc/meta.yaml]: Yes, you're right.

File: recipes/libxc/meta.yaml (hunk 1 above)
Comment (bjodah) [reply recipes/libxc/meta.yaml]: Gotcha, wasn't sure what max_pin was doing. I'm not sure about libxc, but there was max_pin='x.x' in psi4's recipe so I restored that.

Comment (chrisburr) [reply recipes/libxc/meta.yaml]: That seems reasonable to me, for reference:
* If you build libxc 4.3.3:
  - `pin_subpackage('libxc')` becomes: `libxc >=4.3.3,<5`
  - `pin_subpackage('libxc', max_pin='x')` becomes: `libxc >=4.3.3,<5`
  - `pin_subpackage('libxc', max_pin='x.x')` becomes: `libxc >=4.3.3,<4.4`
  - `pin_subpackage('libxc', max_pin='x.x.x')` becomes: `libxc >=4.3.3,<4.3.4`
* If you build libxc 4.1.2:
  - `pin_subpackage('libxc')` becomes: `libxc >=4.1.2,<5`
  - `pin_subpackage('libxc', max_pin='x')` becomes: `libxc >=4.1.2,<5`
  - `pin_subpackage('libxc', max_pin='x.x')` becomes: `libxc >=4.1.2,<4.2`
  - `pin_subpackage('libxc', max_pin='x.x.x')` becomes: `libxc >=4.1.2,<4.1.3`

**Comment by bjodah** [comment]
Thanks for the review

**Pull request merged**